rio_tiler/io/xarray.py

```diff
@@ -72,6 +77,18 @@ def __attrs_post_init__(self):
 
         self.bounds = tuple(self.input.rio.bounds())
         self.crs = self.input.rio.crs
+        if not self.crs:
+            raise MissingCRS(
+                "Dataset doesn't have CRS information, please add it before using rio-tiler (e.g. `ds.rio.write_crs('epsg:4326', inplace=True)`)"
+            )
+
+        if self.crs == WGS84_CRS and (
+            self.bounds[0] < -180
```

Comment:
should then this be ?
```suggestion
            min(self.bounds[0], self.bounds[2]) < -180
```